File: TLBB1.5/Client/Assets/Client/Scripts/Table/Table_Worldboss.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_Worldboss : ITableOperate
{ private const string TAB_FILE_DATA = "worldboss.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_BOSSID,
ID_ZHANWEI,
ID_SCENE_NAME,
ID_SCENE_LIB,
ID_DATE,
ID_COPY_MUSIC,
ID_STARTTIME,
ID_ENDTIME,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private int m_BossID;
 public int BossID { get{ return m_BossID;}}

private string m_CopyMusic;
 public string CopyMusic { get{ return m_CopyMusic;}}

private int m_Date;
 public int Date { get{ return m_Date;}}

private int m_Endtime;
 public int Endtime { get{ return m_Endtime;}}

private string m_SceneLib;
 public string SceneLib { get{ return m_SceneLib;}}

private string m_SceneName;
 public string SceneName { get{ return m_SceneName;}}

private int m_Starttime;
 public int Starttime { get{ return m_Starttime;}}

private int m_Zhanwei;
 public int Zhanwei { get{ return m_Zhanwei;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_Worldboss _values = new Tab_Worldboss();
 _values.m_BossID =  Convert.ToInt32(valuesList[(int)_ID.ID_BOSSID] as string);
_values.m_CopyMusic =  valuesList[(int)_ID.ID_COPY_MUSIC] as string;
_values.m_Date =  Convert.ToInt32(valuesList[(int)_ID.ID_DATE] as string);
_values.m_Endtime =  Convert.ToInt32(valuesList[(int)_ID.ID_ENDTIME] as string);
_values.m_SceneLib =  valuesList[(int)_ID.ID_SCENE_LIB] as string;
_values.m_SceneName =  valuesList[(int)_ID.ID_SCENE_NAME] as string;
_values.m_Starttime =  Convert.ToInt32(valuesList[(int)_ID.ID_STARTTIME] as string);
_values.m_Zhanwei =  Convert.ToInt32(valuesList[(int)_ID.ID_ZHANWEI] as string);

 _hash.Add(nKey,_values); }


}
}

File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_Worldboss.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_Worldboss implements ITableOperate {
 public final static String TAB_FILE_DATA = "/worldboss.txt";
 public static final int ID_BOSSID = 0 ;
public static final int ID_ZHANWEI = 1 ;
public static final int ID_SCENE_NAME = 2 ;
public static final int ID_SCENE_LIB = 3 ;
public static final int ID_DATE = 4 ;
public static final int ID_COPY_MUSIC = 5 ;
public static final int ID_STARTTIME = 6 ;
public static final int ID_ENDTIME = 7 ;
public static final int MAX_RECORD = 8 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private int m_BossID;
 public int GetBossID() { return m_BossID;}

private String m_CopyMusic;
 public String GetCopyMusic() { return m_CopyMusic;}

private int m_Date;
 public int GetDate() { return m_Date;}

private int m_Endtime;
 public int GetEndtime() { return m_Endtime;}

private String m_SceneLib;
 public String GetSceneLib() { return m_SceneLib;}

private String m_SceneName;
 public String GetSceneName() { return m_SceneName;}

private int m_Starttime;
 public int GetStarttime() { return m_Starttime;}

private int m_Zhanwei;
 public int GetZhanwei() { return m_Zhanwei;}

public boolean LoadTable(HashMap<Integer, Table_Worldboss> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_Worldboss> _hash = (HashMap<Integer, Table_Worldboss> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_Worldboss _values = new Table_Worldboss();
 _values.m_BossID =  Integer.parseInt((String)valuesList.get(ID_BOSSID));
_values.m_CopyMusic =  (String)valuesList.get(ID_COPY_MUSIC);
_values.m_Date =  Integer.parseInt((String)valuesList.get(ID_DATE));
_values.m_Endtime =  Integer.parseInt((String)valuesList.get(ID_ENDTIME));
_values.m_SceneLib =  (String)valuesList.get(ID_SCENE_LIB);
_values.m_SceneName =  (String)valuesList.get(ID_SCENE_NAME);
_values.m_Starttime =  Integer.parseInt((String)valuesList.get(ID_STARTTIME));
_values.m_Zhanwei =  Integer.parseInt((String)valuesList.get(ID_ZHANWEI));

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

